python/segyio/utils.py

```diff
@@ -146,6 +146,25 @@ def __repr__(self):
         return msg.format(self.name, self.byte, self.type, self.requires_nonzero_value)
 
 
+class TraceHeaderLayout:
+    def __init__(self, entries):
+        self.entries = list(entries)
+        self.by_byte = {entry.byte: entry for entry in self.entries}
+        self.by_name = {entry.name: entry for entry in self.entries}
```

Comment:
Should the members be marked as private, i.e, prefix `entries, `by_byte`, and `by_name` with `_`?